api/tests/unit/domain/models/CertificationReport_test.js

```diff
@@ -17,6 +19,69 @@ describe('Unit | Domain | Models | CertificationReport', function() {
     });
   });
 
+  describe('#validateForFinalization', function() {
+    it('should validate if all fields are valid', function() {
+      // given
+      const certificationReport = domainBuilder.buildCertificationReport({
+        certificationCourseId: 1,
+        certificationIssueReports: [],
+        hasSeenEndTestScreen: true,
+        isCompleted: true,
+      });
+
+      // when
+      certificationReport.validateForFinalization();
+
+      // then
+      expect(true).to.be.true;
+    });
+
+    // eslint-disable-next-line mocha/no-setup-in-describe
+    [
+      {
+        certificationCourseId: null,
+      },
+      {
+        certificationIssueReports: null,
+      },
+      {
+        hasSeenEndTestScreen: null,
+      },
+      {
+        isCompleted: null,
+      },
+    ].forEach((invalidData) =>
+      it(`should throw an error if ${_getFieldName(invalidData)} is missing`, async function() {
+      // given
+        const certificationReport = new CertificationReport({
+          ...validCertificationReportData, ...invalidData,
+        });
+
+        // when
+        const error = await catchErr(certificationReport.validateForFinalization, certificationReport)();
+
+        // then
+        expect(error).to.be.instanceOf(InvalidCertificationReportForFinalization);
+        expect(error.message).contains(_getFieldName(invalidData));
+      }),
+    );
+
+    it('should throw an error if not completed and abortReason is empty', async function() {
+      // given
+      const certificationReport = new CertificationReport({
+        ...validCertificationReportData,
+        isCompleted: false,
+        abortReason: null,
+      });
+
+      // when
+      const error = await catchErr(certificationReport.validateForFinalization, certificationReport)();
+
+      // then
+      expect(error).to.be.instanceOf(InvalidCertificationReportForFinalization);
```

Comment:
j'ajouterai
expect(error.message).to.equal('Abort reason is required if certificationReport is not completed');
```suggestion
      expect(error).to.be.instanceOf(InvalidCertificationReportForFinalization);
```